synapse/logging/opentracing.py

```diff
@@ -443,7 +443,7 @@ def init_tracer(hs: "HomeServer") -> None:
 
     config = JaegerConfig(
         config=hs.config.tracing.jaeger_config,
-        service_name=f"{hs.config.server.server_name} {hs.get_instance_name()}",
```

Comment:
Did we get any benefit from splitting up the traces by instance or instance type?

Reply:
The only reason I've been holding off from doing this is that we lose the "colouring" of the traces, i.e. it becomes much harder to tell when processing moves to a different process. Not sure if there's another way of achieving the same effect?

Reply:
I'm afraid I would second Erik's sort-of objection here; although having to find the right service is a minor pain, I really find the colouring useful because it helps to spot the points where the processing moves over replication.

Perhaps something else to consider: I wonder if e.g. Grafana's interface for viewing traces is better for finding traces regardless of service?

Reply:
We could also do something like change the service name to the worker *type*? I.e. the instance name without the index. That way it'd be easier to find stuff (as you don't have to figure out the instance), and for the most part instances don't talk to the same type?

Reply:
> We could also do something like change the service name to the worker _type_?

This was going to be my suggestion!

We could also add a tag which is the entire instance name if you did want to filter by that. I think the only time that is useful is if you want to see what happens when an instance is under load or something, but...I don't think this is the right interface for that.

Reply:
just note that 'the instance name without the index' is not a generic Synapse concept; it's our matrix.org Ansible stuff that has this convention. To Synapse, the instance name is opaque and just so happens to include a number.,,

Reply:
BTW +1 to including the real instance name as a tag. I've found it useful to filter out a specific process that was struggling before now. It also gives you a place to look for the right logs when viewing a trace.

Reply:
> The only reason I've been holding off from doing this is that we lose the "colouring" of the traces, i.e. it becomes much harder to tell when processing moves to a different process.

> I really find the colouring useful because it helps to spot the points where the processing moves over replication.

Ahhh, makes sense. I have been mostly looking at the `RoomMessageListRestServlet` which is all turqoise in one process.

For reference, an example that spans multiple processes is `RoomStateEventRestServlet` 

<img width="1023" alt="" src="https://user-images.githubusercontent.com/558581/188932435-e1d9deb4-7844-4351-9c03-ca1223c90ffc.png">


> just note that 'the instance name without the index' is not a generic Synapse concept; it's our matrix.org Ansible stuff that has this convention. To Synapse, the instance name is opaque and just so happens to include a number.,,

Any tips for how to accomplish because this is what I also saw looking at this yesterday and just went with `- Synapse`?

We could strip number suffixes from the opaque worker name. Can we consider that a safe-enough convention?

Reply:
I wouldn't be too offended if we took an educated guess and stripped `"[_-]?[0-9]+$"` off the end of the worker name. I think it's mainly just us/developers that would be using Jaeger anyway, so I don't consider it too harmful.

Reply:
We could make it configurable?

Reply:
I've updated to the sensible default of stripping off the number suffix. Our `matrix.org` worker naming convention seems sane and we can avoid the config.

Reply:
> We could also add a tag which is the entire instance name if you did want to filter by that. I think the only time that is useful is if you want to see what happens when an instance is under load or something, but...I don't think this is the right interface for that.

> BTW +1 to including the real instance name as a tag. I've found it useful to filter out a specific process that was struggling before now. It also gives you a place to look for the right logs when viewing a trace.

Split that out to https://github.com/matrix-org/synapse/pull/13761